src/routes/transactions/handlers/preview.rs

```diff
@@ -0,0 +1,123 @@
+use rocket::futures::future::OptionFuture;
+use serde::{Deserialize, Serialize};
+
+use crate::common::models::addresses::AddressEx;
+use crate::common::models::backend::transactions::SafeTransaction;
+use crate::common::models::data_decoded::{DataDecoded, Operation};
+use crate::providers::info::{DefaultInfoProvider, InfoProvider};
+use crate::routes::contracts::models::DataDecoderRequest;
+use crate::routes::transactions::converters::details::is_trusted_delegate_call;
+use crate::routes::transactions::models::details::TransactionData;
+use crate::routes::transactions::models::TransactionInfo;
+use crate::utils::context::RequestContext;
+use crate::utils::errors::ApiResult;
+use crate::utils::http_client::Request;
+
+#[derive(Deserialize, Debug, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub struct TransactionPreviewRequest {
+    to: String,
+    data: Option<String>,
+    value: String,
+    operation: Operation,
+}
+
+#[derive(Serialize, Debug, PartialEq)]
+#[serde(rename_all = "camelCase")]
+#[cfg_attr(test, derive(Deserialize))]
+pub struct TransactionPreview {
+    pub tx_info: TransactionInfo,
+    pub tx_data: TransactionData,
+}
+
+pub async fn preview_transaction(
```

Comment:
Currently we do all the sub-requests sequentially. Did you check how hard it is to do them in parallel?

Reply:
I didn't, initially for simplicity reasons. I can look into this.

(My preference would be to have this available for the clients as soon as possible so we can do some more testing on `staging`)